### PR TITLE
ta: pkcs11: fixes

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: ta/pkcs11/src/pkcs11_attributes.c [hunk 1]
```diff
@@ -975,12 +975,14 @@ static bool parent_key_complies_allowed_processings(uint32_t proc_id,
...
-	/* Check only if restricted allowed mechanisms list is defined */
-	if (get_attribute_ptr(head, PKCS11_CKA_ALLOWED_MECHANISMS,
-			      (void *)&attr, &size) != PKCS11_CKR_OK) {
+	rc = get_attribute_ptr(head, PKCS11_CKA_ALLOWED_MECHANISMS,
+			       (void *)&attr, &size);
+	if (rc == PKCS11_RV_NOT_FOUND)
 		return true;
-	}
+	if (rc)
+		TEE_Panic("unexpected attributes state");
```
Comment (ruchi393) [inline ta/pkcs11/src/pkcs11_attributes.c:985]: Travis gives error on this -> Argument to TEE_Panic is wrong

pkcs11/src/pkcs11_attributes.c:985:13: error: passing argument 1 of 'TEE_Panic' makes integer from pointer without a cast [-Werror=int-conversion]
   TEE_Panic("unexpected attributes state");
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from out/export-ta_arm32/include/tee_internal_api.h:10:0,
                 from ta/pkcs11/src/pkcs11_attributes.c:12:
out/export-ta_arm32/include/tee_api.h:54:6: note: expected 'TEE_Result {aka unsigned int}' but argument is of type 'const char *'
 void TEE_Panic(TEE_Result panicCode);
      ^~~~~~~~~

Comment (etienne-lms) [reply ta/pkcs11/src/pkcs11_attributes.c:985]: My fault. I messed up when building these commit, I'll fixup...

**Comment by ruchi393** [comment]
Reviewed-by: Ruchika Gupta <ruchika.gupta@linaro.org>

**Comment by etienne-lms** [comment]
tag applied

**Pull request merged**